build-tools/packages/build-cli/src/library/git.ts

```diff
@@ -43,6 +44,52 @@ function getVersionFromTag(tag: string): string | undefined {
 	return ver.version;
 }
 
+/**
+ * Get all the versions for a release group. This function only considers the tags in the repo.
+ *
+ * @param git - The git repository.
+ * @param releaseGroup - The release group to get versions for.
+ * @returns List of version details, or undefined if one could not be found.
+ */
+export async function getVersionsFromTags(
```

Comment:
Consider splitting the "enumerate the tags" and "convert tags to versions" functionality into two functions. Then you can test the tags -> version piece with test data by just passing in an array of tags.

In fact, version-tools has a function called `getVersionsFromStrings` that does the second piece. It takes an array of tag names. It isn't exported but it used to be. I don't remember exactly why it's not exported now - I say just export it and use it.

version-tools also has another function, `getVersions`, that does the first piece. However, that calls git directly, and using SimpleGit is better. I wouldn't export that function.

Summary:

- Create a function with a signature like `getVersionTags(prefix: string): string[]` that returns a list of version tags by calling git.
- Export the `getVersionsFromStrings` function from version-tools.
- Pass the results of `getVersionTags` to `getVersionsFromStrings` to get the array of version strings.
- Ideally add a unit test for the getVersionTags function to version-tools since it's exported now.